src/node/interfaces.cpp

```diff
@@ -859,6 +860,12 @@ class MinerImpl : public Mining
         return TestBlockValidity(state, chainman().GetParams(), chainman().ActiveChainstate(), block, chainman().ActiveChain().Tip(), /*fCheckPOW=*/false, check_merkle_root);
     }
 
+    std::unique_ptr<CBlockTemplate> createNewBlock(const CScript& script_pub_key, bool use_mempool) override
+    {
+        LOCK(::cs_main);
```

Comment:
Is this lock required?

Reply:
Probably not, because `BlockAssembler::CreateNewBlock` takes a lock itself. As does `ChainstateManager::ActiveChainstate()`.

I think I initially added this because `generateblock()` in `rpc/mining.cpp` did it.

I'll add a commit to #30335 to drop it.

@maflcko found something similar

Reply:
(unsure if that was indeed the reason)